### PR TITLE
Improve "name" tests of Promise built-in functions

### LINKED ISSUE #1752: Review name property for the executor functions in Promise.resolve calls
This is feedback after reviewing this [contributed test](https://github.com/tc39/test262/blob/master/implementation-contributed/javascriptcore/stress/private-name-as-anonymous-builtin.js)

- [ ] Review name property for the executor functions in Promise.resolve calls

### PR DESCRIPTION
**Pull request by shvaikalesh**
| Engine              | `name`  | `Promise.allSettled` | CI |
| -------------- | --------- | -------------------- | -- |
| ChakraCore     | 👌          | ❌                               | ❌|
| JSC                  | 👌          | ❌                               | ❌|
| SpiderMonkey | ❌          | ❌                              | ❌|
| V8                    | ❌          | 👌                               | ❌|

Closes #1752.

### CODE REVIEW

File: test/built-ins/Promise/executor-function-name.js [hunk 1]
```diff
@@ -22,3 +23,6 @@ function NotPromise(executor) {
 Promise.resolve.call(NotPromise);
 
 assert.sameValue(Object.prototype.hasOwnProperty.call(executorFunction, "name"), false);
+assert.sameValue(Object.getOwnPropertyNames(executorFunction).join(), "length");
+assert.sameValue(delete executorFunction.name, true);
```
Comment (leobalter) [inline test/built-ins/Promise/executor-function-name.js]: why the delete here after `assert.sameValue(Object.prototype.hasOwnProperty.call(executorFunction, "name"), false);`?

File: test/built-ins/Promise/executor-function-name.js [hunk 2]
```diff
@@ -22,3 +23,6 @@ function NotPromise(executor) {
 Promise.resolve.call(NotPromise);
 
 assert.sameValue(Object.prototype.hasOwnProperty.call(executorFunction, "name"), false);
+assert.sameValue(Object.getOwnPropertyNames(executorFunction).join(), "length");
```
Comment (leobalter) [inline test/built-ins/Promise/executor-function-name.js]: this seems a bit out of scope for this test. This is asserting the only own string-based property names in executorFunction is `"length"` right after you already check that `name` is not an own property of `executorFunction`.

File: test/built-ins/Promise/executor-function-name.js [hunk 3]
```diff
@@ -22,3 +23,6 @@ function NotPromise(executor) {
...
+assert.sameValue(delete executorFunction.name, true);
+assert.sameValue(executorFunction.name, "");
```
Comment (leobalter) [inline test/built-ins/Promise/executor-function-name.js]: this assertion is ok, but the others just seem unnecessary.

File: test/built-ins/Promise/executor-function-name.js (hunk 1 above)
Comment (shvaikalesh) [reply test/built-ins/Promise/executor-function-name.js]: I bet the original test was written at the time when `Function.name` was non-standard and non-configurable.
Totally useless, sorry.

71e1291

File: test/built-ins/Promise/executor-function-name.js (hunk 2 above)
Comment (shvaikalesh) [reply test/built-ins/Promise/executor-function-name.js]: My understanding here is that `Object.getOwnPropertyNames` is used in original test file because it is a stress test and `Object#hasOwnProperty` in hot loop may give incorrect results.

Stress tests are surely out of scope.

71e1291

File: test/built-ins/Promise/all/resolve-element-function-name.js [hunk 4]
```diff
@@ -2,6 +2,7 @@
 // This code is governed by the BSD license found in the LICENSE file.
 
 /*---
+esid: sec-promise.all-resolve-element-functions
 es6id: 25.4.4.1.2
```
Comment (rwaldron) [inline test/built-ins/Promise/all/resolve-element-function-name.js]: When you add an `esid`, you should remove the `es6id`

Comment (shvaikalesh) [reply test/built-ins/Promise/all/resolve-element-function-name.js]: Will do, thank you.
65c230a

File: test/built-ins/Promise/allSettled/reject-element-function-name.js [hunk 5]
```diff
@@ -30,3 +30,4 @@ NotPromise.resolve = function(v) {
 Promise.allSettled.call(NotPromise, [thenable]);
 
 assert.sameValue(Object.prototype.hasOwnProperty.call(rejectElementFunction, 'name'), false);
+assert.sameValue(rejectElementFunction.name, '');
```
Comment (leobalter) [inline test/built-ins/Promise/allSettled/reject-element-function-name.js]: I'm caught from another topic, but if I remember correctly we've had some discussion on anonymous function names in the last TC39 meeting. 

cc @gsathya, mind to remember me if something has changed there?

Comment (gsathya) [reply test/built-ins/Promise/allSettled/reject-element-function-name.js]: The recent change I presented was to make the `name` property an `own` named property on the anonymous function. These tests just seem to be testing that the value of the property is an empty string, not the actual position of the property.

LGTM
